File: Frontend/src/events/converters/battles/enemy.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { EnemyDTO } from "@/events/dto/battles/enemy.dto";
import type { Enemy } from "@/models";
import { VitalConverter } from "@/events/converters/parties/digimons/vital.converter";

export class EnemyConverter {
  public static convert(enemyDto: DeepRequired<EnemyDTO>): Enemy {
    return {
      id: enemyDto.id,
      groupId: enemyDto.groupId,
      condition: enemyDto.condition,
      strength: enemyDto.strength,
      defense: enemyDto.defense,
      speed: enemyDto.speed,
      hp: VitalConverter.convert(enemyDto.hp),
    };
  }
}

File: Backend/Events/Converters/Battles/EnemyConverter.cs
using Backend.Domain.Models.Battles;
using Backend.Events.Converters.Parties.Digimons;
using Backend.Events.DTO.Battles;

namespace Backend.Events.Converters.Battles;

public static class EnemyConverter
{
    public static EnemyDTO ToDTO(Enemy enemy) => new()
    {
        Id = enemy.Id,
        GroupId = enemy.GroupId,
        Condition = enemy.Condition,
        Strength = enemy.Strength,
        Defense = enemy.Defense,
        Speed = enemy.Speed,
        HP = VitalConverter.ToDTO(enemy.HP)
    };
}
